addons/sale/static/src/js/sale_portal_downpayment.js

```diff
@@ -0,0 +1,70 @@
+/** @odoo-module **/
+
+import publicWidget from "web.public.widget";
+
+publicWidget.registry.PortalDownpayment = publicWidget.Widget.extend({
+    selector: '.o_portal_sale_sidebar',
+    events: Object.assign({}, publicWidget.Widget.prototype.events, {
+        'click button[name="o_sale_portal_downpayment_amount_button"]': '_onClickDownpaymentAmountButton',
+        'click button[name="o_sale_portal_full_amount_button"]': '_onClickFullAmountButton',
+    }),
+
+    start: async function () {
+        this.orderDetail = this.$('table#sales_order_table').data();
+        
+        this.payNowButton = this.el.querySelector('#o_sale_portal_paynow')
+        this.fullAmountButton = this.$('button[name="o_sale_portal_full_amount_button"]');
+        this.downpaymentAmountButton = this.$(
+            'button[name="o_sale_portal_downpayment_amount_button"]');
+
+        const params = new URLSearchParams(window.location.search);
+        const comingFromReload = params.has('coming_from_reload');
+        const downpaymentAmount = params.has('downpayment_amount') ?
+            params.get('downpayment_amount') === 'true' : false;
+
+        // Prepare the modal to show if the down payment amount is selected or not.
+        if (downpaymentAmount) {
+            this._onClickDownpaymentAmountButton(false);
+        } else {
+            this._onClickFullAmountButton(false);
+        }
+        
+        // When updating the amount re-open the modal.
+        if (comingFromReload) {
+            this.payNowButton.click();
+        }
+    },
+
+    _onClickDownpaymentAmountButton: function (doReload=true) {
+        this.fullAmountButton.addClass('opacity-50');
+        this.downpaymentAmountButton.removeClass('opacity-50');
+
+        this.$('div[id="o_sale_portal_use_full_amount"]').hide();
+        this.$('div[id="o_sale_portal_use_downpayment_amount"]').show();
+
+        if (doReload) {
+            this._reloadAmount(true)
+        }
+    },
+
+    _onClickFullAmountButton: function(doReload=true) {
+        this.fullAmountButton.removeClass('opacity-50');
+        this.downpaymentAmountButton.addClass('opacity-50');
+
+        this.$('div[id="o_sale_portal_use_full_amount"]').show();
+        this.$('div[id="o_sale_portal_use_downpayment_amount"]').hide();
+
+
+        if (doReload) {
+            this._reloadAmount(false)
+        }
+    },
+
+    _reloadAmount: function (downpayment_amount) {
+        window.location = (
+            `/my/orders/${this.orderDetail.orderId}?access_token=${this.orderDetail.token}&` +
+            `downpayment_amount=${downpayment_amount}&coming_from_reload`
+        );
```

Comment:
@odoo/rd-security could you please verify these lines?

Thank you :)

Reply:
@xmo-odoo `reloadAmount` is to notify the js that we need to reopen the modal after changing the amount (which needs a full reload of the page).